README.md

```diff
@@ -221,6 +221,118 @@ The returned value is taken from the following sources, depending on the item ty
 Return a list with the names of all the fields defined for the item class.
 If an item class doesn't support defining fields upfront, None is returned.
 
+#### class method `get_json_schema(item_class: type) -> dict[str, Any]`
+
+Return a dict with a [JSON Schema](https://json-schema.org/) representation of
+the item class.
+
+The generated JSON Schema reflects field type hints, attribute docstrings and
+class and field metadata of any supported item class. It also supports using
+item classes in field types of other item classes.
+
+For example, given:
+
+```python
+from dataclasses import dataclass
+
+import attrs
+
+
+@dataclass
+class Brand:
+    name: str
+
+@attrs.define
+class Product:
+    name: str
+    """Product name"""
+
+    brand: Brand | None
+    in_stock: bool = True
+```
+
+`ItemAdapter.get_json_schema(Product)` returns:
+
+```python
+{
+    "type": "object",
+    "additionalProperties": False,
+    "properties": {
+        "name": {"type": "string", "description": "Product name"},
+        "brand": {
+            "anyOf": [
+                {"type": "null"},
+                {
+                    "type": "object",
+                    "additionalProperties": False,
+                    "properties": {"name": {"type": "string"}},
+                    "required": ["name"],
+                },
+            ]
+        },
+        "in_stock": {"default": True, "type": "boolean"},
+    },
+    "required": ["name", "brand"],
+}
+```
+
+You can also extend or override JSON Schema data at the item class or field
+level:
+
+-   Set `json_schema_extra` in field metadata to extend or override the JSON
+    Schema data for that field. For example:
+
+    ```python
+    >>> from scrapy.item import Item, Field
+    >>> from itemadapter import ItemAdapter
+    >>> class MyItem(Item):
+    ...     name: str = Field(json_schema_extra={"minLength": 1})
+    ...
+    >>> ItemAdapter.get_json_schema(MyItem)
+    {'type': 'object', 'additionalProperties': False, 'properties': {'name': {'minLength': 1, 'type': 'string'}}, 'required': ['name']}
+
+    ```
+
+-   Define a `__metadata__` class attribute dict and `json_schema_extra` inside
+    to extend or override JSON Schema data for the entire class. For example:
```

Comment:
`__metadata__` is used in [typing.Annotated](https://docs.python.org/3/library/typing.html#typing.Annotated): 

> At runtime, the metadata associated with an Annotated type can be retrieved via the `__metadata__` attribute

Even if they don't directly conflict, I think it'd be better to use a different name. Something like `__itemadapter_metadata__`?

But I might be missing something - is `__metadata__` a standard thing, used for other purposes as well? 

What are other use cases, besides `json_schema_extra`? If it's only `json_schema_extra`, we can also have `__json_schema_extra__` directly, instead of metadata.

Reply:
> Even if they don't directly conflict, I think it'd be better to use a different name. Something like `__itemadapter_metadata__`?

:+1: